### PR TITLE
Correctly insert space ID for reloaded spaces and subnets

### PR DESCRIPTION
**Pull request by nvinuesa**
This patch fixes a bug discovered when reloading spaces on MAAS, where all the discovered subnets were assigned to the alpha space instead of the actual corresponding space.

The bug was simply an incorrectly assigned var during spaces insertion.

## Checklist

- [X] Code style: imports ordered, good names, simple structure, etc
- [X] Comments saying why design decisions were made
- [X] Go unit tests, with comments saying what you're testing
- [ ] ~[Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing~
- [ ] ~[doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

The QA steps must be done in MAAS: after bootstrap and adding a model, `juju spaces` should return the spaces and their subnets correctly assigned.

## Links

**Jira card:** JUJU-

### CODE REVIEW

File: domain/network/service/space.go [hunk 1]
```diff
@@ -327,7 +328,7 @@ func (s *ProviderSpaces) saveSpaces(ctx context.Context, providerSpaces []networ
 			spaceName := network.ConvertSpaceName(string(spaceInfo.Name), spaceNames)
 
 			s.logger.Debugf("Adding space %s from provider %s", spaceName, string(spaceInfo.ProviderId))
-			spaceID, err := s.spaceService.AddSpace(
```
Comment (nvinuesa) [inline domain/network/service/space.go:330]: This was the actual offending line, the `spaceID` var is not assigned to the `spaceID` var defined in line 320 :man_facepalming:

File: domain/network/service/space.go [hunk 2]
```diff
@@ -205,6 +205,7 @@ func (s *ProviderService) saveProviderSubnets(
 		// Add the subnet with the provided space UUID to the upsert list.
 		subnetToUpsert := subnet
 		subnetToUpsert.SpaceID = spaceUUID
+		fmt.Printf("******* spuuid %s\n", spaceUUID)
```
Comment (manadart) [inline domain/network/service/space.go:208]: Oops.

**Review by manadart — APPROVED** [review]
Just remove the printf debugging and we're good.

Rebase your other patches and I will look at those tomorrow.

**Pull request merged**